### ISSUE TITLE
Add  button on noneResultsText 

### ISSUE THREAD

**Issue by kidach** [snapappointments/bootstrap-select#714]
I am using data-live-search = "true"
And its working perfectly.  
I need something like, when noneResultsText is returned, to append button on that, that will add that new result(that is not in options)  as new option 

Is it possible ? :)

And is it possible to have click event on start of opening select.
I wont to populate select when click is fired.
Populate it by ajax. That may be slow, i don't know yet 

I can add 'open' class and open select, but i need event when is clicked  on that DIV mask.

I am not shure how smart is to manualy trigger change event on select

**Comment by melzaiady** [comment]
i use noneResultsText to add button link and create function for on click
```javascript
 $('.selectpicker').selectpicker({
  noneResultsText: 'I found no results <a href="" onclick=(new_client_select()) > Add New Client</a>'
});  
```

**Comment by caseyjhol** [comment]
Closing this in favor of #990. While slightly different functionality, it will be easy enough to implement both at the same time.

**Comment by rish-hyun** [comment]
> i use noneResultsText to add button link and create function for on click
> 
> ```js
>  $('.selectpicker').selectpicker({
>   noneResultsText: 'I found no results <a href="" onclick=(new_client_select()) > Add New Client</a>'
> });  
> ```

Your idea really helped me
Here is my code 
```
$('.selectpicker').selectpicker({
			  noneResultsText: 'No result found <button class="btn btn-light" onclick=(add_opt(this))>Add new option</button>'
		});
		
	function add_opt(event){
		var value = $(event).parents('div').siblings('.bs-searchbox').find('input').val();
		$(event).parents('div').siblings('.selectpicker').append($("<option></option>").text(value)).val(value);
		$('.selectpicker').selectpicker('refresh');
	}
```
